src/main/java/com/exasol/releaserobot/usecases/release/ReleaseInteractor.java

```diff
@@ -33,24 +34,26 @@ public ReleaseInteractor(final ValidateUseCase validateUseCase, final Map<Platfo
     }
 
     @Override
-    public List<Report> release(final UserInput userInput) {
+    public List<Report> release(final UserInput userInput) throws GitHubException {
         final List<Report> reports = new ArrayList<>();
         final Report validationReport = this.validateUseCase.validate(userInput);
         reports.add(validationReport);
         if (!validationReport.hasFailures()) {
             LOGGER.info(() -> "Release started.");
-            final Report releaseReport = this.makeRelease(userInput.getPlatformNames());
+            final Report releaseReport = this.makeRelease(userInput.getRepositoryFullName(),
+                    userInput.getPlatformNames());
             logResults(Goal.RELEASE, releaseReport);
             reports.add(releaseReport);
         }
         return reports;
     }
 
-    private Report makeRelease(final Set<PlatformName> platformNames) {
+    private Report makeRelease(final String repositoryFullName, final Set<PlatformName> platformNames) {
         final Report report = ReportImpl.releaseReport();
         for (final PlatformName platformName : platformNames) {
             try {
-                this.getReleaseMaker(platformName).makeRelease(this.repositoryGateway.getDefaultBranch());
+                this.getReleaseMaker(platformName)
+                        .makeRelease(this.repositoryGateway.getRepositoryWithDefaultBranch(repositoryFullName));
```

Comment:
getting the repository should be out of the loop. Right now we are getting it in every iteration